content/courses/advanced-ada/chapters/limited_types.rst

```diff
@@ -425,6 +431,566 @@ required.
+Limited and nonlimited in full view
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Declaring the full view of a type as limited or nonlimited has implications in
+the way we can use objects of this type in the package body. For example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Limited_Types.Limited_Partial_Full_View
+    :class: ada-expect-compile-error
+
+    package Simple_Recs is
+
+       type Rec_Limited_Full is limited private;
+       type Rec_Nonlimited_Full is limited private;
+
+       procedure Copy (From :        Rec_Limited_Full;
+                       To   : in out Rec_Limited_Full);
+       procedure Copy (From :        Rec_Nonlimited_Full;
+                       To   : in out Rec_Nonlimited_Full);
+
+    private
+
+       type Rec_Limited_Full is limited record
+          I : Integer;
+       end record;
+
+       type Rec_Nonlimited_Full is record
+          I : Integer;
+       end record;
+
+    end Simple_Recs;
+
+    package body Simple_Recs is
+
+       procedure Copy (From :        Rec_Limited_Full;
+                       To   : in out Rec_Limited_Full) is
+       begin
+          To := From;
+          --  ERROR: assignment is forbidden because
+          --         Rec_Limited_Full is limited in
+          --         its full view.
+       end Copy;
+
+       procedure Copy (From :        Rec_Nonlimited_Full;
+                       To   : in out Rec_Nonlimited_Full) is
+       begin
+          To := From;
+          --  OK: assignment is allowed because
+          --      Rec_Nonlimited_Full is
+          --      nonlimited in its full view.
+       end Copy;
+
+    end Simple_Recs;
+
+Here, both :ada:`Rec_Limited_Full` and :ada:`Rec_Nonlimited_Full` are declared
+as :ada:`private limited`. However, :ada:`Rec_Limited_Full` type is limited in
+its full view, while :ada:`Rec_Nonlimited_Full` is nonlimited. As expected,
+the compiler complains about the :ada:`To := From` assignment in the
+:ada:`Copy` procedure for the :ada:`Rec_Limited_Full` type because its full
+view is limited (so no assignment is possible). Of course, in the case of the
+objects of :ada:`Rec_Nonlimited_Full` type, this assignment is perfectly fine.
+
+
+Tagged limited private types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For tagged private types, the partial and full views must match: if a tagged
+type is limited in the partial view, it must be limited in the full view. For
+example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Limited_Types.Tagged_Limited_Private_Types
+
+    package Simple_Recs is
+
+       type Rec is tagged limited private;
+
+    private
+
+       type Rec is tagged limited record
+          I : Integer;
+       end record;
+
+    end Simple_Recs;
+
+Here, the tagged :ada:`Rec` type is limited both in its partial and full views.
+Any mismatch in one of the views triggers a compilation error. (As an
+exercise, you may remove any of the :ada:`limited` keywords from the code
+example and try to compile it.)
```

Comment:
Maybe explain that the rule is for the sake of dynamic dispatching and classwide types. The compiler must not allow any of the types in a derivation class -- the set of types related by inheritance -- to be different regarding assignment and equality (and thus inequality). That's necessary because we are meant to be able to manipulate objects of any type in the entire set of types via the partial view presented by the root type, without knowing which specific tagged type is involved.  [But of course this might not be the place for all this...]

Reply:
Again, you're explanation is perfect, so I'm using it verbatim. Thanks!

However, because it might be too much for the reader at this point — as we haven't discussed any of those more advanced object-oriented topics up to this chapter —, I'm using an admonition for this paragraph.